admissions/views.py

```diff
@@ -1,26 +1,76 @@
-from django.shortcuts import render
+from django.shortcuts import render, redirect
+from django.views.generic import ListView
 
 from OneApply.constants import UserType
 from application.models import HighSchoolApplication
 from register.models import Admin_Staff
 
 
-# TODO: Change this to list view
-def index(request):
-    # TODO: This user ID is hard coded to 1, needs to be changed after sessions are
-    #  implemented
-    user_id = 1
-    applications = get_applications(user_id)
-    context = {
-        "user_type": UserType.ADMIN_STAFF,
-        "constant_ut_student": UserType.STUDENT,
-        "constant_ut_adminStaff": UserType.ADMIN_STAFF,
-        "applications": applications,
-    }
-    return render(request, "admissions/index.html", context)
+class IndexView(ListView):
+    model = HighSchoolApplication
+    paginate_by = 10
+    context_object_name = "applications"
+    template_name = "admissions/index.html"
+
+    def get(self, *args, **kwargs):
+        if not self.request.session.get("is_login", None):
+            return redirect("landingpage:index")
+        return super(IndexView, self).get(*args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["user_type"] = UserType.ADMIN_STAFF
+        context["constant_ut_student"] = UserType.STUDENT
```

Comment:
will this be required?

Reply:
Yes, since the dashboard is being rendered, and there is a condition which uses this variable.

Reply:
Please refrain from updating dashboard's context. That's the bug that plagued Snehal's merge.

Reply:
When I say dashboard is being rendered, I mean to say that we're including our template within the dashboard's template. So when I update the context object it'll have data related to both dashboard and my template. I don't see any other way of sending data to my template, unless it's in a completely different html page itself, and does not include dashboard.html

Reply:
But the dashboard already has these context variables and passing them back to dashboard is something that caused the code to break for the "applications" app. There are a lot of dissimilarities with how the student part is done for the dashboard and frankly, if this is working for you without any bugs then we can revisit them in the next sprint.

Reply:
If I don't send these context variables here, then the page that is rendered doesn't have any context variables. I'm not really sure how you're doing the student part. Could you please check if these context variables are actually present in the dashboard page when you're rendering the student part? Coz each time you render a page, the context gets updated. So I'm not really sure how you're able to send the data you want to send, without updating the context in your case, if that's what you mean.

Reply:
Yes, I believe there are differences in how student and admissions staff sections are handled, so, for now, we can park this given that you tested everything with the latest code off of develop and we can have dashboard as a user story next sprint to investigate any (all) issues that can appear.

Reply:
Ideally speaking we shouldn't have ut_student and ut_adminstaff at all in the URLs. The URL should just say "/dashboard" and the page should render based on the user type who has logged in. So we can revisit this in the later sprint. But that will be a major change affecting both the student and the admin staff dashboard.